File: Python/SelectionSort.py
# time complexity of O(n^2)

arr = [2, 8, 5, 3, 9, 4, 1]

def swap(array, idx0, idx1):
    temp = array[idx0]
    array[idx0] = array[idx1]
    array[idx1] = temp
    return array

def selection_sort(arr):
    for c_min in range(len(arr)):
        for index in range(c_min+1, len(arr)):
            if arr[index] < arr[c_min]:
                arr = swap(arr, index, c_min)
    return arr

print(selection_sort(arr))

File: Java/SelectionSort.java
import java.util.Arrays;

public class SelectionSort
{
    public static void main(String [] args) {
        int[] arr = new int[]{2, 8, 5, 3, 9, 4, 1};

        for(int i=0; i<arr.length;i++) {
            // i refers to the current index of the minimum known element
            for(int j=i+1;j<arr.length;j++) {
                // j refers to the index of the element we are currently comparing the known min to
                if(arr[j] < arr[i])
                    arr = swap(arr, i, j);
            }
        }
        
        System.out.println(Arrays.toString(arr));
    }

    public static int[] swap(int[] arr, int idx0, int idx1) {
        int temp = arr[idx0];
        arr[idx0] = arr[idx1];
        arr[idx1] = temp;

        return arr;
    }
}
